Mathlib/LinearAlgebra/FiniteDimensional.lean

```diff
@@ -1399,27 +1416,37 @@ instance FiniteDimensional.finiteDimensional_subalgebra [FiniteDimensional F E]
   FiniteDimensional.of_subalgebra_toSubmodule inferInstance
 #align finite_dimensional.finite_dimensional_subalgebra FiniteDimensional.finiteDimensional_subalgebra
 
+set_option synthInstance.etaExperiment true in
 instance Subalgebra.finiteDimensional_bot : FiniteDimensional F (⊥ : Subalgebra F E) := by
   nontriviality E
   exact finiteDimensional_of_rank_eq_one Subalgebra.rank_bot
 #align subalgebra.finite_dimensional_bot Subalgebra.finiteDimensional_bot
 
+set_option synthInstance.etaExperiment true in
 theorem Subalgebra.eq_bot_of_rank_le_one {S : Subalgebra F E} (h : Module.rank F S ≤ 1) :
     S = ⊥ := by
   nontriviality E
-  obtain ⟨m, hm, he⟩ := Cardinal.exists_nat_eq_of_le_nat (h.trans_eq Nat.cast_one.symm)
-  haveI := finiteDimensional_of_rank_eq_nat he
+  obtain ⟨m, _, he⟩ := Cardinal.exists_nat_eq_of_le_nat (h.trans_eq Nat.cast_one.symm)
+  haveI I₁ := finiteDimensional_of_rank_eq_nat he
   rw [← not_bot_lt_iff, ← Subalgebra.toSubmodule.lt_iff_lt]
-  haveI := S.toSubmoduleEquiv.symm.finiteDimensional
-  refine' fun hl => (Submodule.finrank_lt_finrank_of_lt hl).not_le (nat_cast_le.1 _)
-  iterate 2 rw [Subalgebra.finrank_toSubmodule, finrank_eq_rank]
-  exact h.trans_eq subalgebra.rank_bot.symm
+  -- Porting FIXME: why does this not pick up the `I₁` when we use
```

Comment:
Does adding an explicit annotation to `I₁` help? I had to do that in lots of places earlier in the file.

Reply:
Yes, it does; applied

Reply:
What do you mean by an "explicit annotation"?

Reply:
I mean `haveI : FiniteDimensional K V :=` instead of `haveI :=`